src/Server.hs

```diff
@@ -0,0 +1,132 @@
+import           ZkFold.Base.Protocol.NonInteractiveProof    (HaskellCore,
+                                                              NonInteractiveProof (Input, Proof, SetupVerify), prove,
+                                                              setupProve, setupVerify)
+import           ZkFold.Base.Protocol.Plonk                  (Plonk (Plonk))
+import           ZkFold.Base.Protocol.Plonkup.Input          (PlonkupInput (..))
+import           ZkFold.Base.Protocol.Plonkup.Proof          (PlonkupProof (..))
+import           ZkFold.Base.Protocol.Plonkup.Prover         (PlonkupProverSecret)
+import           ZkFold.Base.Protocol.Plonkup.Utils          (getParams)
+import           ZkFold.Base.Protocol.Plonkup.Witness        (PlonkupWitnessInput (PlonkupWitnessInput))
+import           ZkFold.Symbolic.Apps.KYC                    (KYCData (KYCData), kycExample)
+import           ZkFold.Symbolic.Compiler
+import           ZkFold.Symbolic.Data.Bool
+import           ZkFold.Symbolic.Data.ByteString             hiding (append, concat)
+import           ZkFold.Symbolic.Data.Combinators            (Iso (..), RegisterSize (Auto))
+import           ZkFold.Symbolic.Interpreter                 (Interpreter)
+
+type N = 18
+
+type K = 8
+
+type F = Zp BLS12_381_Scalar
+
+type Context = (Interpreter F)
+
+data InputData n k c = InputData
+  { statementId :: Int,
+    kycData     :: KYCData n k Auto c
+  }
+  deriving (Generic)
+
+instance FromJSON (InputData N K Context)
+
+instance ToJSON (InputData N K Context)
+
+data OutputData = OutputData
+  { input :: Input (PlonkKYC 29 512),
+    proof :: Proof (PlonkKYC 29 512)
+  }
+  deriving (Generic)
+
+deriving instance Generic (PlonkupInput 1 c)
+
+deriving instance Generic (PlonkupProof c)
+
+instance (ToJSON (ScalarField c), ToJSON (Point c)) => ToJSON (PlonkupProof c)
+
+instance (ToJSON (ScalarField c), ToJSON (Point c)) => ToJSON (PlonkupInput 1 c)
+
+instance (ToJSON (Point BLS12_381_G1)) => ToJSON OutputData
+
+type API = "prove" :> ReqBody '[JSON] (InputData N K Context) :> Post '[JSON] OutputData
+
+type PlonkKYC i n = Plonk U1 i n 1 BLS12_381_G1 BLS12_381_G2 BS.ByteString
+
+kycCheckVerification :: Vector 29 (ScalarField BLS12_381_G1)
+                     -> Fr
+                     -> PlonkupProverSecret BLS12_381_G1
+                     -> (SetupVerify (PlonkKYC 29 512), Input (PlonkKYC 29 512), Proof (PlonkKYC 29 512))
+kycCheckVerification witnessInputs x ps =
+  let Bool ac = compile @Fr (kycExample @N @K @Auto @2) :: Bool (ArithmeticCircuit Fr U1 (((Vector K :*: Vector 1) :*: (Vector 1 :*: Vector N)) :*: (Vector 1 :*: U1)))
+      ac2 = hmap (singleton . unPar1) ac
+      ac3 =
+        hlmap
+          ( \a ->
+              let (q, q1) = splitAt @K @(N + 3) a
+                  (w, w1) = splitAt @N @3 q1
+                  (e, e1) = splitAt @1 @2 w1
+                  (r, t) = splitAt @1 @1 e1
+               in ((q :*: e) :*: (r :*: w)) :*: (t :*: U1)
+          )
+          ac2
```

Comment:
This can be simplified in the latest version of `symbolic-base`. Consult with @TurtlePU. As an example, take a look at the code over here:
https://github.com/zkFold/zkfold-base/blob/vlasin-IVC-update1/symbolic-base/src/ZkFold/Base/Protocol/IVC/Predicate.hs

Reply:
Judging by the type of `PlonkKYC`, first it's needed to bump the version of symbolic-base to the more recent one